slf4j/src/main/scala/org/typelevel/log4cats/slf4j/LoggerFactory.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+package slf4j
+
+import cats.effect.Sync
+import org.slf4j.{Logger => JLogger}
+
+trait LoggerFactory[F[_]] {
```

Comment:
Could this trait be moved to `core`, without the `slf4j` specific methods, and then have a `Slf4jLoggerFactory` that extends it to add `JLogger` methods?

This way, we make log4cats more pallatable for library use, since it doesn't automatically drag in slf4j, and end-users will mostly still instantiate the slf4j specific factory.

It's analogous to what I did in my dead PR. the [core trait](https://github.com/typelevel/log4cats/pull/421/files#diff-188b9221733d2f8a286292dfc94d60f9cbe0d2990b49419ba5b1e0c7af9b8cc4R40), and the [slf4j trait](
https://github.com/typelevel/log4cats/pull/421/files#diff-896fa2ef5da2824bcbf0f8323cdf5f61041b609a3c9006967d9489b01c26409dR25).

Reply:
that is very possible, but I would like some feedback from one of the maintainers for the way forward.

Reply:
I'm not a core contributor of `log4cats`, but I think that's very good advice.